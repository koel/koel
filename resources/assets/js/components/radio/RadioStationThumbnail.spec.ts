import { screen } from '@testing-library/vue'
import { expect, it } from 'vitest'
import UnitTestCase from '@/__tests__/UnitTestCase'
import factory from '@/__tests__/factory'
import Component from './RadioStationThumbnail.vue'

new class extends UnitTestCase {
  protected test () {
    it('renders', () => expect(this.renderComponent().html()).toMatchSnapshot())

    it('emits the clicked event', async () => {
      const { emitted } = this.renderComponent()

      await this.user.click(screen.getByRole('button'))
      expect(emitted().clicked).not.toBeNull()
    })
  }

  private renderComponent (station?: RadioStation) {
    station = station || factory('radio-station', {
      name: 'Beethoven Goes Metal',
      logo: 'https://test/beet.jpg',
    })

    const rendered = this.render(Component, {
      props: {
        station,
      },
    })

    return {
      ...rendered,
      station,
    }
  }
}
