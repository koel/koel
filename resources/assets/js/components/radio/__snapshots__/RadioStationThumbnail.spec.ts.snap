// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`radioStationThumbnail.vue > renders 1`] = `<button style="background-image: url(&quot;undefined/resources/assets/img/covers/default.svg&quot;);" class="thumbnail relative w-full aspect-square bg-no-repeat bg-cover bg-center overflow-hidden rounded-md active:scale-95" data-testid="radio-station-card-thumbnail" title="Play/pause Beethoven Goes Metal"><img alt="Logo" src="https://test/beet.jpg" class="w-full aspect-square object-cover" loading="lazy"><span class="absolute top-0 left-0 w-full h-full group-hover:bg-black/40 no-hover:bg-black/40 z-10"></span><span class="play-icon absolute flex opacity-0 no-hover:opacity-100 items-center justify-center w-[32px] aspect-square rounded-full top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 bg-k-highlight group-hover:opacity-100 duration-500 transition z-20"><br data-testid="Icon" icon="[object Object]" class="ml-0.5 text-k-highlight-fg" size="lg"></span></button>`;
